core/embed/emmc_wrapper/emmc_commands.c

```
@@ -622,183 +794,360 @@
-    // copy over size
-    uint32_t emmc_file_size = file_info.size;
+    if ( path_type == PATH_FILE )
+    {
 
-    // read firmware file to ram
-    uint32_t processed = 0;
-    ExecuteCheck_MSGS_ADV(
-        emmc_fs_file_read(msg_recv.path, 0, bl_buffer->misc_buff, emmc_file_size, &processed), true,
-        {
+        ExecuteCheck_MSGS_ADV(emmc_fs_path_info(path, &file_info), true, {
 #if PRODUCTION
```

Cppcheck (2.10-2)

[error] failed to expand 'ExecuteCheck_MSGS_ADV', it is invalid to use a preprocessor directive as macro parameter

(preprocessorErrorDirective)